### ISSUE TITLE
Undo/Redo

### ISSUE THREAD

**Issue by joex92** [maruohon/litematica#269]
Is there a possibility to hace an Undo/Redo feature?
sometimes when pasting a schematic I mess up so I have to recreate the world to copy the natural generation and paste it where it should have been to do an Undo action...

I believe it could be made by copying the region where a schematic is being pasted, and then when doing Undo its just pasted... idk...

**Comment by maruohon** [comment]
Yes, this has been on my todo list for ages... It will work by making "internal undo schematics" of the affected areas before doing the actual operation.

**Comment by SimplyAG** [comment]
Support for undo/redo features

**Comment by zisis912** [comment]
are you ever going to implement this?

**Comment by maruohon** [comment]
@zisis912 Yes.

(Longer answer: There is a LOT of other stuff that needs to be rewritten and cleaned up first before any real new features get added. And there are a LOT of other new features fighting for development time to be realized. And my available modding time recently went down a lot after I got a job/learning position. So no idea when it will get added.)

**Comment by zisis912** [comment]
oh okay, I am looking forward to seeing these new features then!

**Comment by el97** [comment]
This feature will be fantastic to use when it gets implemented.

**Comment by mattilpea** [comment]
still waiting
